src/components/EditGoals.js

```diff
@@ -21,7 +21,9 @@ class EditGoals extends React.Component {
     const { intl } = props;
 
     this.defaultType = 'yearlyBudget';
-    this.state = { goals: [...props.goals || {}] };
+    this.state = { goals: [...(props.goals || []).map(g => {
+      return { ...g };
+    }) ] };
```

Comment:
My poor - small - tired - brain can't process that :-)

Reply:
Yeah it's a bit ugly isn't it?
I've done better ASCII art in my life.

Basically just solving the problem of `this object is not extensible`. So need to make a copy. But spreading `props.goals` is not enough, because each `goal` is not extensible. A better idea?

Reply:
I'll use `cloneDeep` from `lodash`, cleaner.